### PR TITLE
add absolute component

### PR DESCRIPTION
**Pull request by elizfunk**
resolves issue #318

### CODE REVIEW

File: .storybook/Absolute.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      <Absolute top={8} left={0}>
+        <Flag>Hello Flag</Flag>
+      </Absolute>
+      <Image src="https://www.priceline.com/home/public/assets/images/photos/photo-aruba.jpg" />
+    </RelativeBox>
+  ))
+  .add('Positioning an icon', () => (
+    <Box p={3}>
+      <RelativeCard p={4}>
+        <Text mt="8px" align="justify">
```
Comment (jxnblk) [inline .storybook/Absolute.js]: This can be `mt={2}`

File: .storybook/Absolute.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
+import React from 'react'
+import { storiesOf } from '@storybook/react'
+import styled from 'styled-components'
+import { Absolute, Box, Card, Flag, Flex, Icon, Image, Text } from '../src'
+
+const RelativeBox = styled(Box)`
+  position: relative;
```
Comment (jxnblk) [inline .storybook/Absolute.js]: Would be good to add a `Relative` component in a follow up PR as discussed in #318

File: src/__tests__/Absolute.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      .create(<Absolute top={10} left={0} zIndex={2} />)
+      .toJSON()
+    expect(json).toMatchSnapshot()
+    expect(json).toHaveStyleRule('top', '10px')
+    expect(json).toHaveStyleRule('left', '0px')
+    expect(json).toHaveStyleRule('z-index', '2')
+  })
+
+  test('renders with flag and Child', () => {
+    const json2 = renderer.create(
```
Comment (jxnblk) [inline src/__tests__/Absolute.js]: Since this is scoped in the function, the variable can be `json` here as well

File: .storybook/Absolute.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+const ExtraLargeAbsolute = styled(Absolute)`
+  height: 400px;
+`
+
+storiesOf('Absolute', module)
+  .add('Over an image', () => (
+    <RelativeBox width={1 / 2}>
+      <Absolute top={8} left={0}>
+        <Flag>Hello Flag</Flag>
+      </Absolute>
```
Comment (jxnblk) [inline .storybook/Absolute.js]: NB/FWIW: many times negative margins can achieve the same thing as positioning with fewer chances of CSS bugs

**Comment by jxnblk** [comment]
Thanks for putting this together! LGTM :shipit:

File: .storybook/Absolute.js (hunk 2 above)
Comment (elizfunk) [reply .storybook/Absolute.js]: That's my next PR : )

File: src/__tests__/Absolute.js (hunk 3 above)
Comment (elizfunk) [reply src/__tests__/Absolute.js]: Good point.  Will fix.

File: .storybook/Absolute.js (hunk 1 above)
Comment (elizfunk) [reply .storybook/Absolute.js]: Will update.

Comment (elizfunk) [reply .storybook/Absolute.js]: Updated

File: src/__tests__/Absolute.js (hunk 3 above)
Comment (elizfunk) [reply src/__tests__/Absolute.js]: Updated

File: docs/Absolute.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+
+# Absolute
+
+Use `<Absolute />` component to control position.
+This component is an extension of the `<Box />` component.
+
+```.jsx
+<Card>
```
Comment (hakimelek) [inline docs/Absolute.md]: Maybe make it more explicit that `Card` here should be `Relative`.  I would also think that a `Relative` component should come with the `Absolute` component. The change of position here will be relative to body, which I don't know if it's a behavior that a consumer would like seeing while wrapping it in a `Card`.

File: src/Absolute.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  PropTypes.string,
+  PropTypes.array
+])
+
+Absolute.propTypes = {
+  top: numberStringOrArray,
+  bottom: numberStringOrArray,
+  left: numberStringOrArray,
+  right: numberStringOrArray,
+  zIndex: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
```
Comment (hakimelek) [inline src/Absolute.js]: I like the API 👍

File: .storybook/Absolute.js (hunk 2 above)
Comment (hakimelek) [reply .storybook/Absolute.js]: I would vote that `Relative` should be part of this PR.

Comment (jxnblk) [reply .storybook/Absolute.js]: We discussed in the meeting and think it's fine for Relative to be a follow up PR

**Pull request merged**